ballerina/modules/parser/parser.bal

```diff
@@ -104,10 +103,56 @@ public class Parser {
     isolated function createOperationNode(string name, RootOperationType kind, Location location)
     returns OperationNode|Error {
         OperationNode operation = new(name, kind, location);
+        Token token = check self.peekNextNonSeparatorToken();
+        if (token.kind == T_OPEN_PARENTHESES) {
+            check self.addVariableDefinition(operation);
+        }
         check self.addSelections(operation);
         return operation;
     }
 
+    isolated function addVariableDefinition(OperationNode operationNode) returns Error? {
+        Token token = check self.readNextNonSeparatorToken(); // Read the open parantheses here
+        while (token.kind != T_CLOSE_PARENTHESES) {
+            token = check self.readNextNonSeparatorToken();
+            if (token.kind != T_DOLLAR) {
```

Comment:
Let's do not add parentheses around the simple conditional expressions. Check the other places as well.

```suggestion
            if token.kind != T_DOLLAR {
```